### DISCUSSION TITLE
APK Packaging issue

### DISCUSSION THREAD

**Discussion by Maskedviper** [flet-dev/flet#2910] [Q&A]
### Question

We are working on creating an app using Python and the Flet framework, after following several videos and links we are still running into an error in packaging the code as an apk file using the Flet framework.When we add the libraries one at a time they modulenotfounderror Is removed for the previous library not included but when we include all libraries in one go its taking hours for the packaging process and it still doesn't end up packaging.

We were getting the error "(ModuleNotFoundError: No module named 'fcntl')", indicating that fcntl is not available on windows and its specific to Unix. So I used a Windows Subsystem for Linux (WSL) and made sure all the environment variables `ANDROID_SDK_ROOT` and `NDK_VERSION` are set correctly. Then running the p4a command "p4a apk --requirements=python3,flet,requests,beautifulsoup4,pandas,numpy,matplotlib,scikit-learn,xgboost --arch=arm64-v8a,armeabi-v7a,x86_64 --sdk-dir=$ANDROID_SDK_ROOT --ndk-dir=$ANDROID_SDK_ROOT/ndk/$NDK_VERSION --dist-name=mydist --package=org.example.myapp --name "My Application" --version 0.1 --bootstrap=sdl2"

We have also specified the correct NDK VERSION and we still seem to be running into errors

### Error message

```bash
root@DESKTOP-13SP0P1:/mnt/c/Users/AQ/Documents/Extracted UI Updated# export ANDROID_SDK_ROOT="/mnt/c/Users/AQAppData/Local/Android/Sdk"
root@DESKTOP-13SP0P1:/mnt/c/Users/AQ/Documents/Extracted UI Updated# ls "$ANDROID_SDK_ROOT/cmdline-tools/latest/bin/sdkmanager"
ls: cannot access '/mnt/c/Users/AQAppData/Local/Android/Sdk/cmdline-tools/latest/bin/sdkmanager': No such file or directory
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by Xiang990293** [comment]
Same issue here. I have no clue.
